### ISSUE TITLE
Support Unhandled exception on watchOS

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#2747]
### Description

As pointed out in https://github.com/getsentry/sentry-cocoa/issues/406#issuecomment-1453147037 crash reports are not possible on watchOS. Instead, we can offer unhandled exception reporting with only the exception stack trace. We could investigate grabbing the stack traces of other threads.

**Comment by brustolin** [comment]
> We could investigate grabbing the stack traces of other threads

According to https://github.com/getsentry/sentry-cocoa/issues/406#issuecomment-1171872518 is not possible to get the stack traces of other threads.

**Comment by armcknight** [comment]
That is true @brustolin, we can't get the exact state of the other threads at the moment of the crash, but we can at least reliably grab the backtrace from the exception handler, which would be better than nothing. 

We could at least test to see what it looks like to [grab the other threads like bugsnag is doing anyways](https://github.com/bugsnag/bugsnag-cocoa/pull/1403/files#diff-baacd50226c787d2a378bbb0abe179059997aa14dd982ad60129df6b4f3a5e5fR136), but it won't be exact.
